starport/pkg/cosmosanalysis/cosmosanalysis.go

```diff
@@ -72,6 +73,71 @@ func FindImplementation(modulePath string, interfaceList []string) (found []stri
 	return found, nil
 }
 
+// FindStructFields finds the fields from a
+// specific struct declared into the app
+func FindStructFields(
```

Comment:
I think we should parse the proto files instead of `x/`

We need the proto definition of the type since we use it as a field for the message. Currently, if the proto definition doesn't exist, the generated app doesn't compile because we suppose it is defined in protos

Parsing proto files would allows to return a proper error

Reply:
I made the ast parser think about the use case of the SPM, we can scaffold the proto messages by starport, but we move the messages from some files to others. In this case, we need to read all proto files and check all message structures. We need to make some benchmarks, but I think ast can be more quickly them the proto analysis 3thr party package.

Reply:
But the implementation is almost the same. We can easily move to proto analysis....

Reply:
> I made the ast parser think about the use case of the SPM

What do you mean by the use case of SPM?

I'm not really thinking about the implementation detail, using Go at may be a good solution. My only concern is that we can use structures from Go source code but the code generation will fail if this structure is not defined in proto. It would be better to have an explicit error if the type is not defined in proto